src/vs/workbench/parts/terminal/electron-browser/terminalActions.ts

```diff
@@ -358,8 +360,11 @@ export class SplitTerminalAction extends Action {
 			if (!path) {
 				return Promise.resolve(void 0);
 			}
-			this._terminalService.splitInstance(instance, path);
-			return this._terminalService.showPanel(true);
+			return instance.getCwd(this._terminalService.configHelper).then(cwd => {
+				(path as IShellLaunchConfig).cwd = cwd;
```

Comment:
Oh no, I added a `PromiseLike<any>` 😱. Can we type the promise as `Promise<IShellLaunchConfig>` to avoid this?